README.md

```diff
@@ -1,63 +1,3280 @@
+
+Accessing a customer's device brings with it a set of responsibilities, both
+technically and ethically. Device support assumes a level of competence and
+understanding by a customer for a support agent, and as such support agents
+should ensure that these levels are met successfully.
+
+Additionally, it may not always be possible to access the device directly,
+especially if the VPN component isn't working.
+
+The following two sections lay down groundwork for both responsibilites and
+accessing a device via another gateway device.
+
+### 3.1 Key Responsibilities
+There are some key points which should be followed to ensure that we are never
+destructive when supporting a customer:
+
+* Always ask permission before carrying out non-read actions. This includes
+    situations such as stopping/restarting/starting services which are otherwise
+    functional (such as the Supervisor). This is *especially* important in cases
+    where this would stop otherwise functioning applications (such as stopping
+    balenaEngine)
+* Ensure that the customer is appraised of any non-trivial non-read actions that
+    you are going to take before you carry those actions out on-device. If they
+    have given you permission to do 'whatever it takes' to get the device
+    running again, you should still pre-empt your actions by communicating this
+    clearly
+* During the course of carrying out non-read actions on a device, should the
+    customer be required to answer a query before being able to proceed, make it
+    clear to them what you have already carried out, and that you need a
+    response before continuing. Additionally ensure that any incoming agents
+    that may need to access the device have all of your notes and actions up
+    to this point, so they can take over in your stead
+* *Never* reboot a device without permission, especially in cases where it
+    appears that there is a chance that the device will not recover (which may
+    be the case in situations where the networking is a non-optimal state). It
+    is imperative in these cases that the customer is made aware that this could
+    be an outcome in advance, and that they must explicitly give permission for
+    a reboot to be carried out
+
+### 3.3 Device Access via a Gateway Device
+
+In the examples, we're able to stay connected to the device when the OpenVPN
+service isn't running purely because we're using a debug image because the SSH
+port allows passwordless root connections. Had we been running a production
+device, then the device would have been in the 'Offline' state, but it would
+still potentially have had network access and be otherwise running correctly.
+This brings up an issue though, how can we connect to a faulty production device
+in the field?
+
+The answer comes from the mechanism behind how SSH is tunneled through the VPN,
+and we can actually use another device (in the 'Online' state) on the same
+local network as an 'Offline' device to do this
```

Comment:
maybe we should add some notes on using/checking the "device heart beat" API that @richbayliss deployed in recent months to validate that the device is indeed online (communicating with the api) but the VPN is failing?

Reply:
@richbayliss What do you think? Do you have a quick one-liner `curl` we could run from the device to see when its last heartbeat was (and what state it denotes the device in)? You could of course just look at the OpenVPN service too, but it'd be nice to have as a 'double confirmation'. :)